core/src/main/java/org/apache/iceberg/actions/SizeBasedFileRewriter.java

```diff
@@ -258,6 +261,24 @@ protected long writeMaxFileSize() {
     return (long) (targetFileSize + ((maxFileSize - targetFileSize) * 0.5));
   }
 
+  /**
+   * returns already set outputSpecId rewriter will use, value is set by user using `output-spec-id`
+   * see more {@link #outputSpecId(Map)}
+   */
+  protected int outputSpecId() {
+    return outputSpecId;
+  }
+
+  private int outputSpecId(Map<String, String> options) {
+    int specId =
+        PropertyUtil.propertyAsInt(options, RewriteDataFiles.OUTPUT_SPEC_ID, table.spec().specId());
+    Preconditions.checkArgument(
+        table.specs().containsKey(specId),
+        "Output spec id %s is not a valid spec id for table",
```

Comment:
Generally we follow the pattern

"Cannot x because y. (Possible remediation z)"

So
"Cannot use output spec id %d because the table does not contain a reference to this spec-id." 

Or something like that